server/packetio.go

```diff
@@ -76,7 +77,8 @@ func (p *packetIO) setReadTimeout(timeout time.Duration) {
 	p.readTimeout = timeout
 }
 
-func (p *packetIO) readOnePacket() ([]byte, error) {
+// maxAllowedPacket is the maximum size of one packet. accumulatedLength count the length of received data totally.
+func (p *packetIO) readOnePacket(maxAllowedPacket uint64, accumulatedLength uint64) ([]byte, error) {
```

Comment:
How about making `maxAllowedPacket` and `accumulatedLength` the member fields of `packetIO`, just like `readTimeout`?

So that we do not need to change the signature of `readOnePacket`.